lib/authorizer.ts

```diff
@@ -0,0 +1,75 @@
+import * as crypto from 'crypto';
+import { ExpressRequest } from './interfaces';
+
+const debugLog = process.env.ENABLE_LOGENTRIES_WEBHOOK_AUTH_LOGGING
+  ? console.log
+  : (message) => {};
+
+export class Authorizer {
+  password: string;
+
+  constructor(password: string, dependencies?: any) {
+    this.getFromAuthorizationHeader = this.getFromAuthorizationHeader.bind(this);
+    this._checkHash = this._checkHash.bind(this);
+
+    if (!password) throw new Error(`'password' is required for logentries webhook auth.`);
+    this.password = password;
+
+    if (dependencies == null) { dependencies = {}; }
+  }
```

Comment:
If we were starting from scratch I don't think I'd use an object oriented approach.  

You'll notice a lot of this constructor is boilerplate sets up state that used nowhere.  For instance, lines 12 and 13 set up a `this` binding for the member functions, but `getFromAuthorizationHeader` does not reference `this` at all, anywhere.  `_checkHash` references `this` to access the password that the middleware was initialized with, but you'll notice the middleware itself has next to no logic in it besides constructing and calling this Authorizer object.  This could have simply been an inline function that acquires the password value via closure.  

The last parameter of the constructor accepts an optional `dependencies` object... which is thrown away?  Perhaps they had some roadmap in mind where this would eventually be relevant, but now its just noise.

Reply:
Agreed, OO wouldn't have been my first approach to this either.

In any case, I can pretty quickly get rid of the useless boilerplate in the constructor without changing the structure of the code.

ff1a7cb296ce5e01c7ee9b3e1fd04f53c6e661da